app/screens/integration_selector/user_list_row/index.tsx

```diff
@@ -0,0 +1,190 @@
+        if (!isShared(user)) {
+            return null;
+        }
+
+        return (
+            <ChannelIcon
+                name={user.username}
+                type={General.DM_CHANNEL}
+                isActive={false}
+                isArchived={false}
+                isUnread={true}
+                isInfo={true}
+                size={18}
+                shared={true}
+            />
+        );
+    };
+
+    const intl = useIntl();
+    const {username} = user;
+    const userID = user.id;
+    const style = getStyleFromTheme(theme);
+
+    let usernameDisplay = `@${username}`;
+    if (isMyUser) {
+        usernameDisplay = intl.formatMessage({
+            id: 'mobile.more_dms.you',
+            defaultMessage: '@{username} - you',
+        }, {username});
+    }
+
+    const showTeammateDisplay = teammateNameDisplay !== username;
+    const itemTestID = `${testID}.${userID}`;
+    const displayUsernameTestID = `${testID}.display_username`;
+    const profilePictureTestID = `${itemTestID}.profile_picture`;
+    const userIcon = renderIcon() || (<></>);
+
+    return (
+        <View style={style.container}>
+            <CustomListRow
+                id={userID}
+                onPress={onPressRow}
+                enabled={enabled}
+                selectable={selectable}
+                selected={selected}
+                testID={testID}
+            >
+                <View style={style.profileContainer}>
+                    <ProfilePicture
+                        author={user}
+                        size={32}
+                        iconSize={24}
+                        testID={profilePictureTestID}
+                    />
+                </View>
+                <View
+                    style={style.textContainer}
+                    testID={itemTestID}
+                >
+                    <View>
+                        <View style={style.indicatorContainer}>
+                            <Text
+                                style={style.username}
+                                ellipsizeMode='tail'
+                                numberOfLines={1}
+                                testID={displayUsernameTestID}
+                            >
+                                {usernameDisplay}
+                            </Text>
+                            <BotTag
+                                show={Boolean(user.is_bot)}
+                            />
+                            <GuestTag
+                                show={isGuest(user.roles)}
+                            />
+                        </View>
+                    </View>
+                    {showTeammateDisplay &&
+                        <View>
+                            <Text
+                                style={style.displayName}
+                                ellipsizeMode='tail'
+                                numberOfLines={1}
+                            >
+                                {teammateNameDisplay}
```

Comment:
I'm not sure what `teammateNameDisplay` means. I propagate it from the AutocompleteSelector because it's being observed there, but not sure what it does or why.

Reply:
shouldn't be that way? I'm not sure I understand what this `teammateDisplay`means :-/

Reply:
I'd like to leave it this way and see once we start the v2 design, I don't know if the design is the same for both user list rows.

Reply:
LGTM keeping these two components separated for the time being, but you handle teammateNameDisplay correctly.